### DISCUSSION TITLE
code.pybricks.com editor monaco documentation

### DISCUSSION THREAD

**Discussion by BertLindeman** [pybricks/support#1367] [Q&A] [answered]
The python editor used in code.pybricks.com (and beta I think) is monaco.

Is there documentation somewhere about the tools that are built-in that editor?
I also looked on microsoft but could not find a simple user guide for the editor.

As an example: I know that you can use **TAB** to extra indent a selected part of the source, and **SHIFT**-**TAB** to decrease the indent.

A function I was looking for was to make a comment / uncomment of a selected part of the source. 
And related to #1300 if it would be possible to make the standard indent be 2 bytes in stead of 4. So less scrolling would be needed.

Would be nice to know where to look for the doc.

**Comment by dlech** [comment, marked as answer]
Monaco is the editor used in VS Code, so the VS Code docs and source code are effectively the Monaco docs. So technically, it can do anything that VS Code can do, however many of these features are not enabled or would require additional programming to replicate parts of VS Code that aren't part of the core editor.

One thing you can do is press <kbd>F1</kbd> to bring up the command palette to see what features are actually there. Then you can type to search commands. For example, here are the commands related co comments.

![image](https://github.com/pybricks/support/assets/963645/2895d906-7715-42a2-bd29-41aa405f6f6c)

**Reply by BertLindeman** [reply]
Wow, David, 

Exactly what I wanted to know.

And even the tabsize can be adapted:

![afbeelding](https://github.com/pybricks/support/assets/8142081/ebd4c63e-130e-4334-9d00-00b45d0edc95)


Thanks
